src/main/scala/cromwell/binding/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -91,16 +101,55 @@ trait WdlStandardLibraryFunctions extends WdlFunctions[WdlValue] {
    */
   protected def read_boolean(params: Seq[Try[WdlValue]]): Try[WdlBoolean] =
     read_string(params) map { s => WdlBoolean(java.lang.Boolean.parseBoolean(s.value.trim.toLowerCase)) }
-  protected def write_lines(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("write_lines")
+
+  private def writeToTsv(params: Seq[Try[WdlValue]], wdlClass: Class[_ <: WdlValue with TsvSerializable]) = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      downcast <- Try(wdlClass.cast(singleArgument))
+      tsvSerialized <- downcast.tsvSerialize
+      file <- writeContent(wdlClass.getSimpleName.toLowerCase, tsvSerialized)
+    } yield file
+  }
+
+  protected def write_lines(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    writeToTsv(params, classOf[WdlArray])
+  }
+
+  protected def write_map(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    writeToTsv(params, classOf[WdlMap])
+  }
+
+  protected def write_object(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    writeToTsv(params, classOf[WdlObject])
+  }
+
+  protected def write_objects(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    writeToTsv(params, classOf[WdlArray])
+  }
+
+  protected def glob(params: Seq[Try[WdlValue]]): Try[WdlArray] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      globVal = singleArgument.valueString
+      files = interface.glob(globPath(globVal), globVal)
+      wdlFiles = files map { WdlFile(_, isGlob = false) }
+    } yield WdlArray(WdlArrayType(WdlFileType), wdlFiles toSeq)
+  }
+
   protected def write_tsv(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("write_tsv")
-  protected def write_map(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("write_map")
-  protected def write_object(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("write_object")
-  protected def write_objects(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("write_objects")
   protected def write_json(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("write_json")
-  protected def glob(params: Seq[Try[WdlValue]]): Try[WdlArray] = fail("glob")
 }
 
-class NoFunctions extends WdlStandardLibraryFunctions
+class NoFunctions extends WdlStandardLibraryFunctions with IoInterface {
+  override def readFile(path: String): String = throw new NotImplementedError()
+  override def writeFile(path: String, content: String): Unit = throw new NotImplementedError()
+  override def listContents(path: String): Iterable[String] = throw new NotImplementedError()
+  override def glob(path: String, pattern: String): Seq[String] = throw new NotImplementedError()
+  override def writeTempFile(path: String, prefix: String, suffix: String, content: String): String = throw new NotImplementedError()
+  override def exists(path: String): Boolean = throw new NotImplementedError()
+  override def interface: IoInterface = throw new NotImplementedError()
+  override def isValidPath(path: String): Boolean = throw new NotImplementedError()
```

Comment:
Personally I view them as the same although in the last few days multiple folks have expressed that ??? seems like more of a placeholder than an intentional thing. That said, if one's going to have those exceptions, to go along with the making it explicit angle I'd like to have a message along the lines of "foo is not currently supported".